packages/@internationalized/number/src/NumberParser.ts

```diff
@@ -180,34 +200,108 @@ class NumberParserImpl {
   }
 
   sanitize(value: string) {
+    let sanitizedValue = value.trim();
+
+    let numeralMatches = sanitizedValue.match(NUMERALS_REGEX);
+    if (numeralMatches && this.options.numberingSystem !== 'arab') {
+      let firstNumeralMatch = numeralMatches[0];
+      let lastNumeralMatch = numeralMatches[numeralMatches.length - 1];
+      let beforeAbs = sanitizedValue.slice(0, sanitizedValue.indexOf(firstNumeralMatch));
+      let afterAbs = sanitizedValue.slice(sanitizedValue.lastIndexOf(lastNumeralMatch) + 1);
+      let abs = sanitizedValue.slice(sanitizedValue.indexOf(firstNumeralMatch), sanitizedValue.lastIndexOf(lastNumeralMatch) + 1);
+
+      // Replace group and decimal symbols with the current locale's symbols
+      let decimalSymbolMatch = abs.match(DECIMAL_SYMBOLS_REGEX);
+      let groupSymbolMatch = abs.match(GROUPING_SYMBOLS_REGEX);
+      let integerPart: string;
+      let decimalPart: string;
+      if (decimalSymbolMatch) {
+        let firstDecimalSymbol = decimalSymbolMatch[0];
+        let lastDecimalSymbol = decimalSymbolMatch[decimalSymbolMatch.length - 1];
+        integerPart = abs.slice(0, abs.lastIndexOf(lastDecimalSymbol));
+        decimalPart = abs.slice(abs.lastIndexOf(lastDecimalSymbol) + 1, abs.length);
+        integerPart = integerPart.replace(GROUPING_SYMBOLS_REGEX, '');
+        let isArabic = NUMERALS_ARAB_REGEX.test(abs);
+        let isHanidec = NUMERALS_HANIDEC_REGEX.test(abs);
+        if (isArabic) {
+          // Replace Arabic numerals with Latin numerals,
+          // then parse the integer part to remove leading zeros,
+          // and finally replace Latin numerals with Arabic numerals.
+          integerPart = (
+            parseInt(
+              integerPart
+              .replace(NUMERALS_ARAB_REGEX, (d) => NUMERALS_ARAB.indexOf(d).toString()),
+              10
+            ).toString()
+            .replace(NUMERALS_REGEX, (d) => NUMERALS_ARAB.split('')[parseInt(d, 10)])
+          );
+        } else if (isHanidec) {
+          // Replace Hanidec numerals with Latin numerals,
+          // then parse the integer part to remove leading zeros,
+          // and finally replace Latin numerals with Hanidec numerals.
+          integerPart = (
+            parseInt(
+              integerPart
+              .replace(NUMERALS_HANIDEC_REGEX, (d) => NUMERALS_HANIDEC.indexOf(d).toString()),
+              10
+            ).toString()
+            .replace(NUMERALS_REGEX, (d) => NUMERALS_HANIDEC.split('')[parseInt(d, 10)])
+          );
+        } else {
+          integerPart = parseInt(integerPart, 10).toString();
+        }
+        let decimalSymbol = decimalSymbolMatch.length > 1 && lastDecimalSymbol === firstDecimalSymbol ? '' : lastDecimalSymbol;
```

Comment:
wouldn't the parsing up above have failed if there were multiple decimal symbols?

I'm generally confused by what is happening inside this `if (decimalSymbol !== '')`

Reply:
A number can't have more than one decimal symbol, so if it does, it's probably a grouping symbol, in which case we set `decimalSymbol = ''`.

For example: "1,234,567,890" or "1.234.567.890". 

Within `if (decimalSymbol !== '')`, if the last decimal symbol does not match the decimal symbol for the current locale, I'm testing the following cases to determine whether to replace the symbol in the string with the decimal symbol for the current locale.

1. If the `integerPart`, before the `lastDecimalSymbol`, is longer than 3 digits. i.e. "1234,56" would become "1234.56", however but "12,234" is ambiguous and should remain unchanged.
2. If the `integerPart` is zero, we assume a decimal symbol. "0,123" becomes "0.123".
3. If it has both decimal and group symbols, but they're swapped. "123.456,789" becomes "123456.789"
4. If the 'decimalPart' is longer than 3 characters, "12,3456789" becomes "12.3456789"